zebrad/src/components/tokio.rs

```diff
@@ -23,3 +25,34 @@ impl TokioComponent {
         })
     }
 }
+
+/// Zebrad's handler for various signals
+async fn signal_handler() -> Result<(), Report> {
+    tokio::signal::ctrl_c().await?;
+    Ok(())
+}
+
+/// Extension trait to centralize entry point for runnable subcommands that
+/// depend on tokio
+pub(crate) trait RuntimeRun {
+    fn run(&mut self, fut: impl Future<Output = Result<(), Report>>);
+}
+
+impl RuntimeRun for Runtime {
+    fn run(&mut self, fut: impl Future<Output = Result<(), Report>>) {
+        let result = self.block_on(async move {
+            tokio::select! {
+                result = fut => result,
+                result = signal_handler() => result,
+            }
+        });
+
+        match result {
+            Ok(()) => {}
+            Err(e) => {
+                eprintln!("Error: {:?}", e);
+                std::process::exit(1);
+            }
+        }
+    }
+}
```

Comment:
I changed it so that we now have support for graceful shutdown via ctrl-c, that way all the destructors will get called like normal so there's no need for special case cleanup logic.